### ISSUE TITLE
osm verify connectivity return incorrect result 

### ISSUE THREAD

**Issue by addozhang** [flomesh-io/osm-edge#107]
**Bug description**:

Using the command `osm verify connectivity` to verify pod accessibility, the result shows the failure below while pod is accessible. 

```shell
 osm verify connectivity --from-pod curl/curl-54bf6469ff-crwds --to-pod httpbin/httpbin-c8b4bfc5c-kctrd --to-service httpbin
---------------------------------------------
[+] Context: Verify if pod "curl/curl-54bf6469ff-crwds" can access pod "httpbin/httpbin-c8b4bfc5c-kctrd" for service "httpbin/httpbin"
Status: Failure
Reason: A verification step failed
Suggestion: Please follow the suggestions listed in the failed steps below to resolve the issue

[++] Context: Verify if namespace "curl" is monitored
Status: Failure
Reason: Missing label "openservicemesh.io/monitored-by" on namespace "curl"
Suggestion: Add label "openservicemesh.io/monitored-by" on namespace "curl" to include it in the mesh and restart the app

[++] Context: Verify if namespace "httpbin" is monitored
Status: Success

[++] Context: Verify Envoy sidecar on pod "curl/curl-54bf6469ff-crwds"
Status: Failure
Reason: Did not find Envoy sidecar on pod "curl/curl-54bf6469ff-crwds"
Suggestion: Ensure pod "curl/curl-54bf6469ff-crwds" has sidecar injection enabled

[++] Context: Verify Envoy sidecar on pod "httpbin/httpbin-c8b4bfc5c-kctrd"
Status: Success

[++] Context: Verify Envoy config for traffic:
	source pod: curl/curl-54bf6469ff-crwds
	source service: <nil>
	destination pod: httpbin/httpbin-c8b4bfc5c-kctrd
	destination service: httpbin/httpbin
	destination protocol: http
Status: Failure

[+++] Context: Verify Envoy config on source
Status: Unknown
Reason: Error retrieving Envoy config for pod "curl/curl-54bf6469ff-crwds", err: "Pod curl-54bf6469ff-crwds in namespace curl is not a part of a mesh"

---------------------------------------------
```

```shell
kubectl exec "$(kubectl get pod -n curl -l app=curl -o jsonpath='{.items..metadata.name}')" -n curl -- curl -sI http://httpbin.httpbin:14001/get
HTTP/1.1 200 OK
server: gunicorn
date: Fri, 21 Oct 2022 19:27:39 GMT
content-type: application/json
content-length: 267
access-control-allow-origin: *
access-control-allow-credentials: true
osm-stats-namespace: httpbin
osm-stats-kind: Deployment
osm-stats-name: httpbin
osm-stats-pod: httpbin-c8b4bfc5c-kctrd
connection: keep-alive
```

**Affected area (please mark with X where applicable)**:
- Install                    [ ]
- SMI Traffic Access Policy  [ ]
- SMI Traffic Specs Policy   [ ]
- SMI Traffic Split Policy   [ ]
- Permissive Traffic Policy  [ ]
- Ingress                    [ ]
- Egress                     [ ]
- Sidecar Control Plane      [ ]
- CLI Tool                   [x]
- Metrics                    [ ]
- Certificate Management     [ ]
- Sidecar Injection          [ ]
- Logging                    [ ]
- Debugging                  [ ]
- Tests                      [ ]
- Demo                       [ ]
- CI System                  [ ]

**Expected behavior**: 

curl pod is out of mesh and httpbin one is inside of mesh. There is one AccessControl policy applied:

```yaml
kubectl get accesscontrol httpbin -n httpbin -o yaml
apiVersion: policy.openservicemesh.io/v1alpha1
kind: AccessControl
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"policy.openservicemesh.io/v1alpha1","kind":"AccessControl","metadata":{"annotations":{},"name":"httpbin","namespace":"httpbin"},"spec":{"backends":[{"name":"httpbin","port":{"number":14001,"protocol":"http"}}],"sources":[{"kind":"Service","name":"curl","namespace":"curl"}]}}
  creationTimestamp: "2022-10-14T06:49:07Z"
  generation: 1
  name: httpbin
  namespace: httpbin
  resourceVersion: "2246510"
  uid: ca333e5e-498f-4bde-b9e6-58f906c0dab3
spec:
  backends:
  - name: httpbin
    port:
      number: 14001
      protocol: http
  sources:
  - kind: Service
    name: curl
    namespace: curl
status:
  currentStatus: committed
  reason: successfully committed by the system
```

**Steps to reproduce the bug (as precisely as possible)**:

Execute command `osm verify connectivity --from-pod curl/curl-54bf6469ff-crwds --to-pod httpbin/httpbin-c8b4bfc5c-kctrd --to-service httpbin`.

**How was osm-edge installed?**:

Installed with osm CLI.

**Anything else we need to know?**:

**Bug report archive**:

**Environment**: Ubuntu 20.04
- osm-edge version (use `osm version`): v1.23.8+k3s2
- Kubernetes version (use `kubectl version`): k3s v1.23.8+k3s2
- Size of cluster (number of worker nodes in the cluster): 2
- Others:
